telemetry/metrics.go

```diff
@@ -14,6 +15,21 @@ import (
 	"github.com/prometheus/common/expfmt"
 )
 
+// globalTelemetryEnabled is a private variable that stores the telemetry enabled state.
+// It is set on initialization and does not change for the lifetime of the program.
+var globalTelemetryEnabled atomic.Bool
```

Comment:
Good use of global var but why do you use an atomic bool when it is not modified after init? Could be a simple bool.